src/python/WMComponent/JobAccountant/AccountantWorker.py

```diff
@@ -474,38 +475,42 @@ def handleJob(self, jobID, fwkJobReport):
             elif jobType == "Express" and set(outputMap).difference(outputModules) == {'write_RAWRAW'}:
                 pass
             else:
-                failJob = True
+                # any job that is not multi-step and Processing/Production must FAIL!
                 if jobType in ["Processing", "Production"]:
-                    cmsRunSteps = 0
-                    for step in fwkJobReport.listSteps():
-                        if step.startswith("cmsRun"):
-                            cmsRunSteps += 1
-                    if cmsRunSteps > 1:
-                        failJob = False
-
-                if failJob:
-                    jobSuccess = False
-                    logging.error("Job %d , list of expected outputModules does not match job report, failing job",
-                                  jobID)
-                    logging.debug("Job %d , expected outputModules %s", jobID, sorted(outputMap.keys()))
-                    logging.debug("Job %d , fwjr outputModules %s", jobID, sorted(outputModules))
-                    fileList = fwkJobReport.getAllFilesFromStep(step='logArch1')
+                    cmsRunSteps = sum([1 for step in fwkJobReport.listSteps() if step.startswith("cmsRun")])
+                    if cmsRunSteps == 1:
```

Comment:
It is unclear to me if this logic is correct. For instance, what if job had only one step and it was completed. Then the above cmsRunSteps sum will be equal to 1 while this condition will reject it. I'm not saying it is the case but from reading the code it can be, since
```
>>> sum([1])
1
```

Reply:
Yes, that's possible and it would mean that a given job didn't evaluate to True in this `if` statement:
https://github.com/dmwm/WMCore/blob/e4c212a647708fb89659dca2c4a592efec4a1d2c/src/python/WMComponent/JobAccountant/AccountantWorker.py#L465

meaning that something has been wrongly constructed in the workflow/job, and that it should indeed be marked as failed.